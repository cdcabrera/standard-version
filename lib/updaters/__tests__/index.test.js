const { resolveUpdaterObjectFromArgument } = require('../index');

describe('updaters', () => {
  it('should attempt to run updaters with an invalid format', () => {
    const mockWarn = jest.fn();
    const spy = jest.spyOn(console, 'warn').mockImplementation(mockWarn);

    expect(() => resolveUpdaterObjectFromArgument('invalid updater')).toThrowErrorMatchingSnapshot(
      'invalid updater, string'
    );
    expect(mockWarn.mock.calls).toMatchSnapshot('console messages');

    spy.mockClear();
  });

  it('should attempt to run updaters with other formats', () => {
    const mockWarn = jest.fn();
    const spy = jest.spyOn(console, 'warn').mockImplementation(mockWarn);

    const mockValidUpdater = resolveUpdaterObjectFromArgument({
      readVersion: Function.prototype,
      writeVersion: Function.prototype,
      mockUpdater: 'Lorem ipsum mock'
    });

    const mockUpdaterString = resolveUpdaterObjectFromArgument({
      readVersion: Function.prototype,
      writeVersion: Function.prototype,
      updater: 'Lorem ipsum mock'
    });

    const mockUpdaterType = resolveUpdaterObjectFromArgument({
      readVersion: Function.prototype,
      writeVersion: Function.prototype,
      type: 'json'
    });

    const mockUpdaterFilename = resolveUpdaterObjectFromArgument({
      readVersion: Function.prototype,
      writeVersion: Function.prototype,
      filename: 'Lorem ipsum filename'
    });

    expect({
      mockValidUpdater,
      mockUpdaterString,
      mockUpdaterType,
      mockUpdaterFilename
    }).toMatchSnapshot('updaters');

    expect(mockWarn.mock.calls).toMatchSnapshot('console messages');

    spy.mockClear();
  });
});
