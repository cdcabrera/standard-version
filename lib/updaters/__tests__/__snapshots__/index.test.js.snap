// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updaters should attempt to run updaters with an invalid format: console messages 1`] = `
Array [
  Array [
    "Unable to obtain updater for: \\"invalid updater\\"
 - Error: Unsupported file (invalid updater) provided for bumping.
 Please specify the updater \`type\` or use a custom \`updater\`.
 - Skipping...",
  ],
]
`;

exports[`updaters should attempt to run updaters with an invalid format: invalid updater, string 1`] = `"Cannot read property 'readVersion' of undefined"`;

exports[`updaters should attempt to run updaters with other formats: console messages 1`] = `Array []`;

exports[`updaters should attempt to run updaters with other formats: updaters 1`] = `
Object {
  "mockUpdaterFilename": Object {
    "filename": "Lorem ipsum filename",
    "readVersion": [Function],
    "writeVersion": [Function],
  },
  "mockUpdaterString": Object {
    "readVersion": [Function],
    "updater": "Lorem ipsum mock",
    "writeVersion": [Function],
  },
  "mockUpdaterType": Object {
    "readVersion": [Function],
    "type": "json",
    "writeVersion": [Function],
  },
  "mockValidUpdater": Object {
    "mockUpdater": "Lorem ipsum mock",
    "readVersion": [Function],
    "writeVersion": [Function],
  },
}
`;
